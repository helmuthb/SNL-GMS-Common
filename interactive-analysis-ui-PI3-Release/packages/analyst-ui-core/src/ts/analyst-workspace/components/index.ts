export { ReduxApolloWaveformDisplay as WaveformDisplay } from './waveform-display';
export { LoadingScreen } from './loading-screen';
export { ReduxApolloEventList as EventList } from './event-list';
export { ReduxApolloSignalDetectionList as SignalDetectionList } from './signal-detection-list';
export { ReduxApolloWorkflow as Workflow } from './workflow';
export { ReduxApolloMap as Map } from './map';
export { ReduxApolloAzimuthSlowness as AzimuthSlowness } from './azimuth-slowness';
