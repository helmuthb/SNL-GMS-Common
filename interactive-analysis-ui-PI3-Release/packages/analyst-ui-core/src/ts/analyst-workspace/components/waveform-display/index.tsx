import * as React from 'react';
import { compose, graphql } from 'react-apollo';
import * as ReactRedux from 'react-redux';

import { AnalystWorkspaceState } from '../../state';
import * as Actions from '../../state/actions';
import { createDetectionMutation, rejectDetections, updateDetectionsMutation } from './graphql/mutations';
import { waveformDisplayQuery, WaveformDisplayQueryInput } from './graphql/queries';
import { WaveformDisplay, WaveformDisplayReduxProps } from './waveform-display';
import { WaveformSortType } from './waveform-display-controls';

// map parts of redux state into this component as props
const mapStateToProps = (state: AnalystWorkspaceState): Partial<WaveformDisplayReduxProps> => ({
    currentTimeInterval: state.app.currentTimeInterval,
    currentOpenEventHypId: state.app.openEventHypId,
    selectedSdIds: state.app.selectedSdIds,
    waveformSortType: state.app.selectedSortType
});

// map actions dispatch callbacks into this component as props
const mapDispatchToProps = (dispatch): Partial<WaveformDisplayReduxProps> => ({
    setSelectedSdIds: (ids: string[]) => {
        dispatch(Actions.setSelectedSdIds(ids));
    },
    setSelectedSortType: (selectedSortType: WaveformSortType) => {
        dispatch(Actions.setSelectedSortType(selectedSortType));
    }
});

/**
 * higher-order component react-redux(react-apollo(WaveformDisplay))
 */
export const ReduxApolloWaveformDisplay: React.ComponentClass<Pick<{}, never>> = compose(
    ReactRedux.connect(mapStateToProps, mapDispatchToProps),
    graphql(waveformDisplayQuery, {
        options: (props: WaveformDisplayReduxProps) => {
            const skip = props.currentTimeInterval === undefined;
            const variables: WaveformDisplayQueryInput | {} = skip ? {}
                : {
                    timeRange: {
                        startTime: String(props.currentTimeInterval.startTimeSecs),
                        endTime: String(props.currentTimeInterval.endTimeSecs),
                    },
                    distanceToSourceInput: {
                        sourceId: props.currentOpenEventHypId,
                        sourceType: 'Event'
                    }
                };
            // work-around to only fetch based on props, if the current time interval is undefined
            return {
                variables,
                fetchPolicy: skip ? 'cache-only' : undefined
            };
        },
    }),
    graphql(createDetectionMutation, { name: 'createDetection' }),
    graphql(updateDetectionsMutation, { name: 'updateDetections' }),
    graphql(rejectDetections, { name: 'rejectDetectionHypotheses' })
)(WaveformDisplay);

export { WaveformDisplay, WaveformDisplayProps } from './waveform-display';
