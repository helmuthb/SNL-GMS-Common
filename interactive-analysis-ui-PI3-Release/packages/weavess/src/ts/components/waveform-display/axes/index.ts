export { TimeAxis } from './x-axis';
export { YAxis } from './y-axis';
