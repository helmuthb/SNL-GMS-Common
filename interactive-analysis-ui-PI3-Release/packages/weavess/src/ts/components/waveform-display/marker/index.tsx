export { MarkerObjects } from './marker';
export { SelectionWindow } from './selection-window';
export { MoveableMarker } from './moveable-marker';
export { VerticalMarker, VerticalMarkerObject } from './vertical-marker';
