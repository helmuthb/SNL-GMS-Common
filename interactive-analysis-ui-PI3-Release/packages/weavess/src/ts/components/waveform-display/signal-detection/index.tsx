import * as React from 'react';

export interface SignalDetectionProps {
    id: string;
    channelId: string;

    position: number;
    label: string;
    color: string;
    timeSecs: number;

    predicted?: boolean;
    selectedSignalDetections: string[] | undefined;

    getTimeSecsForClientX(clientX: number): number | undefined;
    toggleDragIndicator(show: boolean, color: string): void;
    positionDragIndicator(clientX: number): void;
    onClick?(e: React.MouseEvent<HTMLDivElement>, sdId: string): void;
    onContextMenu?(e: React.MouseEvent<HTMLDivElement>, channelId: string, sdId?: string): void;
    onSignalDetectionDragEnd?(sdId: string, timeSecs: number | undefined): void;
}

export class SignalDetection extends React.Component<SignalDetectionProps, {}> {

    private detectionWrapperRef: HTMLDivElement | null;
    private detectionLineRef: HTMLDivElement | null;

    public render() {
        const color = this.props.color;
        const isSelected = this.props.selectedSignalDetections &&
            this.props.selectedSignalDetections.indexOf(this.props.id) > -1;
        const pickStyle: React.CSSProperties = {
            borderLeft: '1px solid ' + color,
            bottom: this.props.predicted ? '10%' : '55%',
            // tslint:disable-next-line:no-magic-numbers
            left: `${this.props.position * 100}%`,
            position: 'absolute',
            top: this.props.predicted ? '55%' : '10%',
            boxShadow: isSelected ? `0px 0px 10px 3px ${color}` : 'initial'
        };
        const pickLabelStyle: React.CSSProperties = {
            bottom: this.props.predicted ? '10%' : 'initial',
            left: `calc(4px + ${this.props.position} * 100%)`,
            lineHeight: '80%',
            position: 'absolute',
            top: this.props.predicted ? 'initial' : '10%',
            fontSize: '0.95rem',
            color,
            fontWeight: 200
        };
        return (
            <div
                ref={ref => this.detectionWrapperRef = ref}
                onMouseDown={this.onMouseDown}
            >
                <div
                    ref={ref => this.detectionLineRef = ref}
                    style={pickStyle}
                />
                <div
                    className="weavess-sd-label"
                    onClick={this.onClick}
                    onContextMenu={this.onContextMenu}
                    style={pickLabelStyle}
                >
                    {this.props.label}
                </div>
            </div>
        );
    }

    public componentDidUpdate(prevProps: SignalDetectionProps) {
        if (!this.detectionLineRef) return;

        // if the color changes, flash animation
        if (prevProps.color !== this.props.color && this.props.color !== 'red') {
            this.detectionLineRef.style.borderColor = 'white';
            setTimeout(() => {
                if (!this.detectionLineRef) return;

                this.detectionLineRef.style.borderColor = this.props.color;
                this.detectionLineRef.style.transition = 'border-color 0.5s ease-in';
                setTimeout(() => {
                    if (!this.detectionLineRef) return;
                    this.detectionLineRef.style.transition = null;
                    // tslint:disable-next-line:no-magic-numbers align
                }, 500);
                // tslint:disable-next-line:no-magic-numbers align
            }, 500);
        }
    }

    private readonly onClick = (e: React.MouseEvent<HTMLDivElement>): void => {
        // prevent propagation of these events so that the underlying channel click doesn't register
        e.stopPropagation();
        if (this.props.onClick) {
            this.props.onClick(e, this.props.id);
        }
    }

    private readonly onContextMenu = (e: React.MouseEvent<HTMLDivElement>): void => {
        e.stopPropagation();
        if (this.props.onContextMenu) {
            this.props.onContextMenu(e, this.props.channelId, this.props.id);
        }
    }

    private readonly onMouseDown = (e: React.MouseEvent<HTMLDivElement>): void => {
        // if context-menu, don't trigger
        if (e.button === 2) return;
        // prevent propagation of these events so that the underlying channel click doesn't register
        e.stopPropagation();
        const start = e.clientX;
        let currentPos = e.clientX;
        let isDragging = false;
        let diff = 0;

        const onMouseMove = (event: MouseEvent) => {
            if (!this.detectionWrapperRef) return;

            currentPos = event.clientX;
            diff = Math.abs(currentPos - start);
            // begin drag if moving more than 1 pixel
            if (diff > 1 && !isDragging) {
                isDragging = true;
                this.detectionWrapperRef.style.filter = 'brightness(0.5)';
                this.props.toggleDragIndicator(true, this.props.color);
            }
            if (isDragging) {
                this.props.positionDragIndicator(currentPos);
            }
        };

        const onMouseUp = (event: MouseEvent) => {
            if (!this.detectionWrapperRef) return;

            if (isDragging) {
                this.props.toggleDragIndicator(false, this.props.color);
                this.detectionWrapperRef.style.filter = 'initial';
                const time = this.props.getTimeSecsForClientX(currentPos);
                if (this.props.onSignalDetectionDragEnd) {
                    this.props.onSignalDetectionDragEnd(this.props.id, time);
                }
            }
            document.body.removeEventListener('mousemove', onMouseMove);
            document.body.removeEventListener('mouseup', onMouseUp);
        };

        document.body.addEventListener('mousemove', onMouseMove);
        document.body.addEventListener('mouseup', onMouseUp);
    }
}
