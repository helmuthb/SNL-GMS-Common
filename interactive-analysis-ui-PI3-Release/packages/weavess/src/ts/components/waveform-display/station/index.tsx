import * as React from 'react';

import { forEach } from 'lodash';
import * as Entities from '../../../entities';
import { Channel } from './channel';

/**
 * Station Props
 */
export interface StationProps {

    stationConfig: Entities.StationConfig;

    displayStartTimeSecs: number;
    displayEndTimeSecs: number;
    workerRpcs: any[];
    selectedSignalDetections: string[] | undefined;
    selectedChannels: string[] | undefined;
    expanded: boolean;
    isMeasureWindow: boolean;

    // hotkeys
    amplitudeScaleHotKey: string;
    amplitudeScaleSingleResetHotKey: string;

    // callbacks
    toggleExpansion(stationId: string);
    setYAxisBounds(channelId: string, min: number, max: number);
    getViewRange(): [number, number];
    onLoad(): void;

    onMouseMove(e: React.MouseEvent<HTMLDivElement>, xPct: number): void;
    onMouseDown(e: React.MouseEvent<HTMLDivElement>, xPct: number): void;
    onMouseUp(e: React.MouseEvent<HTMLDivElement>, xPct: number, channelId: string, timeSecs: number): void;
    onContextMenu?(e: React.MouseEvent<HTMLDivElement>, channelId: string, sdId?: string): void;
    onChannelLabelClick?(e: React.MouseEvent<HTMLDivElement>, channelName: string): void;
    onSignalDetectionClick?(e: React.MouseEvent<HTMLDivElement>, sdId: string): void;
    onSignalDetectionDragEnd?(sdId: string, timeSecs: number): void;
    onKeyPress?(e: React.KeyboardEvent<HTMLDivElement>, clientX: number,
        clientY: number, channelId: string, timeSecs: number): void;
    updateMeasureWindow(measureWindowSelection: Entities.MeasureWindowSelection): void;
    onMaskClick?(event: React.MouseEvent<HTMLDivElement>, channelId: string, maskId: string[]): void;
}

/**
 * Station State
 */
// tslint:disable-next-line:no-empty-interface
export interface StationState {
}

/**
 * A station consisting of a default channel and 0 or more non-default channels with data
 */
export class Station extends React.Component<StationProps, StationState> {

    /**
     * the default props for all stations, if not passed in
     */
    public static defaultProps: Entities.StationDefaultConfig = {
        height: 75
    };

    /**
     * A ref to the default channel
     */
    public defaultChannelRef: Channel | null;

    /**
     * An object containing refs to all additional channels
     */
    public additionalChannelRefs: { [id: string]: Channel | null } = {};

    public constructor(props: StationProps) {
        super(props);
    }

    // ************************************
    // BEGIN REACT COMPONENT LIFECYCLE METHODS
    // ************************************

    /**
     * React component lifecycle
     */
    public render() {

        const singleHeight = this.props.stationConfig.height || Station.defaultProps.height;
        const totalHeight = this.props.expanded && this.props.stationConfig.nonDefaultChannels ?
            singleHeight * (this.props.stationConfig.nonDefaultChannels.length + 1)
            : singleHeight;
        // console.log("state height: " + this.state.height);
        return (
            <div
                style={{
                    height: totalHeight,
                    display: 'flex',
                    flexDirection: 'column',
                    flex: '1 0 auto'
                }}
            >
                <Channel
                    ref={ref => this.defaultChannelRef = ref}
                    key={this.props.stationConfig.defaultChannel.id}

                    stationId={this.props.stationConfig.id}
                    subChannel={false}
                    height={this.props.stationConfig.height || Station.defaultProps.height}
                    channelConfig={this.props.stationConfig.defaultChannel}

                    displayStartTimeSecs={this.props.displayStartTimeSecs}
                    displayEndTimeSecs={this.props.displayEndTimeSecs}
                    workerRpcs={this.props.workerRpcs}
                    selectedSignalDetections={this.props.selectedSignalDetections}
                    selectedChannels={this.props.selectedChannels}
                    isMeasureWindow={this.props.isMeasureWindow}

                    // hotkeys
                    amplitudeScaleHotKey={this.props.amplitudeScaleHotKey}
                    amplitudeScaleSingleResetHotKey={this.props.amplitudeScaleSingleResetHotKey}

                    // callbacks
                    setYAxisBounds={this.props.setYAxisBounds}
                    getViewRange={this.props.getViewRange}
                    onLoad={this.props.onLoad}
                    onContextMenu={this.props.onContextMenu}
                    onChannelLabelClick={this.props.onChannelLabelClick}
                    onSignalDetectionClick={this.props.onSignalDetectionClick}
                    onSignalDetectionDragEnd={this.props.onSignalDetectionDragEnd}
                    onKeyPress={this.props.onKeyPress}
                    onMouseMove={this.props.onMouseMove}
                    onMouseDown={this.props.onMouseDown}
                    onMouseUp={this.props.onMouseUp}
                    updateMeasureWindow={this.props.updateMeasureWindow}
                    onMaskClick={this.props.onMaskClick}
                />
                {
                    this.props.expanded && this.props.stationConfig.nonDefaultChannels ?
                        (
                            this.props.stationConfig.nonDefaultChannels.map(channelConfig => (
                                <Channel
                                    key={channelConfig.id}
                                    ref={ref => this.additionalChannelRefs[channelConfig.id] = ref}

                                    stationId={this.props.stationConfig.id}
                                    subChannel={true}
                                    height={this.props.stationConfig.height || Station.defaultProps.height}
                                    channelConfig={{ ...channelConfig }}

                                    displayStartTimeSecs={this.props.displayStartTimeSecs}
                                    displayEndTimeSecs={this.props.displayEndTimeSecs}
                                    workerRpcs={this.props.workerRpcs}
                                    selectedSignalDetections={this.props.selectedSignalDetections}
                                    selectedChannels={this.props.selectedChannels}
                                    isMeasureWindow={false}

                                    // hotkeys
                                    amplitudeScaleHotKey={this.props.amplitudeScaleHotKey}
                                    amplitudeScaleSingleResetHotKey={this.props.amplitudeScaleSingleResetHotKey}

                                    // callbacks
                                    setYAxisBounds={this.props.setYAxisBounds}
                                    getViewRange={this.props.getViewRange}
                                    onLoad={this.props.onLoad}
                                    onContextMenu={this.props.onContextMenu}
                                    onChannelLabelClick={this.props.onChannelLabelClick}
                                    onSignalDetectionClick={this.props.onSignalDetectionClick}
                                    onSignalDetectionDragEnd={this.props.onSignalDetectionDragEnd}
                                    onKeyPress={this.props.onKeyPress}
                                    onMouseMove={this.props.onMouseMove}
                                    onMouseDown={this.props.onMouseDown}
                                    onMouseUp={this.props.onMouseUp}
                                    updateMeasureWindow={this.props.updateMeasureWindow}
                                    onMaskClick={this.props.onMaskClick}
                                />
                            ))
                        )
                        : []
                }
            </div>
        );
    }

    /**
     * React component lifecycle
     */
    public componentDidUpdate(prevProps: StationProps) {
        const delay = 100;
        if (prevProps.expanded !== this.props.expanded) {
            window.setTimeout(this.props.onLoad, delay);
        }
    }

    // ************************************
    // END REACT COMPONENT LIFECYCLE METHODS
    // ************************************

    /**
     * Reset the amplitude to the default.
     */
    public resetAmplitude = () => {
        if (this.defaultChannelRef) {
            this.defaultChannelRef.resetAmplitude();
        }
        if (this.additionalChannelRefs) {
            forEach(
                this.additionalChannelRefs,
                // tslint:disable-next-line
                channel => (channel) ? channel.resetAmplitude() : null);
        }
    }
}
