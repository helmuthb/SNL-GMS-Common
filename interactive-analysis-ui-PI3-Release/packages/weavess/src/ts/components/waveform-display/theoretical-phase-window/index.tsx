import * as React from 'react';

/**
 * Props for TheoreticalPhaseWindow
 */
export interface TheoreticalPhaseWindowProps {
    left: number;
    right: number;
    label: string;
    color: string;
}

/**
 * Displays a window of time on a channel where a phase may theoretically exist
 */
export class TheoreticalPhaseWindow extends React.Component<TheoreticalPhaseWindowProps, {}> {

    /**
     * React component lifecycle
     */
    public render() {
        const fracToPct = 100;
        const color = this.props.color;
        const theoreticalWindowStyle: React.CSSProperties = {
            backgroundColor: color,
            bottom: '20%',
            left: `${this.props.left * fracToPct}%`,
            opacity: 0.35,
            position: 'absolute',
            right: `${(1 - this.props.right) * fracToPct}%`,
            top: '55%',
        };
        const labelStyle: React.CSSProperties = {
            bottom: '5%',
            color: this.props.color,
            fontSize: '65%',
            left: `${this.props.left * fracToPct}%`,
            position: 'absolute',
            right: `${(1 - this.props.right) * fracToPct}%`,
            textAlign: 'center',
        };

        return (
            <div>
                <div style={theoreticalWindowStyle}/>
                <div style={labelStyle}>
                    {this.props.label}
                </div>
            </div>
        );
    }
}
