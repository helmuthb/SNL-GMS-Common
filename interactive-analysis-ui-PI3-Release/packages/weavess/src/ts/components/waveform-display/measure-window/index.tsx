import { NonIdealState } from '@blueprintjs/core';
import * as React from 'react';
import { MeasureWindowSelection, StationConfig } from '../../../entities';
import { TimeAxis } from '../axes';
import { Station } from '../station';
import { StationLabel } from '../station/label';

/**
 * MeasureWindow Props
 */
export interface MeasureWindowProps {
    heightPx: number;
    workerRpcs: any[]; // webWorkers used to load waveforms into graphics space
    selectedSignalDetections: string[] | undefined;
    waveformDisplayRef: HTMLDivElement | null;

    // hotkeys
    amplitudeScaleHotKey: string;
    amplitudeScaleSingleResetHotKey: string;

    // callbacks
    toggleExpansion(stationId: string);
    onLoad(): void;
    onMouseMove(e: React.MouseEvent<HTMLDivElement>, xPct: number): void;
    onMouseDown(e: React.MouseEvent<HTMLDivElement>, xPct: number): void;
    onMouseUp(e: React.MouseEvent<HTMLDivElement>, xPct: number, channelId: string, timeSecs: number): void;
    onContextMenu?(e: React.MouseEvent<HTMLDivElement>, channelId: string, sdId?: string): void;
    onChannelLabelClick?(e: React.MouseEvent<HTMLDivElement>, channelName: string): void;
    onSignalDetectionClick?(e: React.MouseEvent<HTMLDivElement>, sdId: string): void;
    onSignalDetectionDragEnd?(sdId: string, timeSecs: number): void;
    onKeyPress?(e: React.KeyboardEvent<HTMLDivElement>, clientX: number,
        clientY: number, channelId: string, timeSecs: number): void;
    onMaskClick?(event: React.MouseEvent<HTMLDivElement>, channelId: string, maskId: string[]): void;

}

/**
 * MeasureWindow State
 */
export interface MeasureWindowState {
    measureWindowSelection?: MeasureWindowSelection;
    yAxisBounds?: [number, number];
    xAxisBounds?: [number, number];
    measureWindowHeightPx: number;
}

/**
 * MeasureWindow Component
 */
export class MeasureWindow extends React.Component<MeasureWindowProps, MeasureWindowState> {

    /**
     * Reference to the containing div.
     */
    public containerRef: HTMLDivElement | null;

    /**
     * Reference to the station.
     */
    public stationRef: Station | null;

    /**
     * Reference to the station label.
     */
    public stationLabelRef: StationLabel | null;

    /**
     * Reference to the measureWindowDiv.
     */
    public measureWindowRef: HTMLDivElement | null;

    /**
     * Constructor
     */
    public constructor(props: MeasureWindowProps) {
        super(props);
        this.state = {
            measureWindowSelection: undefined,
            yAxisBounds: undefined,
            xAxisBounds: undefined,
            measureWindowHeightPx: this.props.heightPx
        };
    }

    /**
     * React component lifecycle
     */
    public render() {
        const TIME_AXIS_HEIGHT = 35;

        if (!this.state.measureWindowSelection ||
            !this.state.measureWindowSelection.channelRef) {
            return (
                <div
                    ref={ref => this.containerRef = ref}
                    style={{
                        flex: '0 0 auto',
                        borderBottom: '0.25rem solid',
                        height: `${this.state.measureWindowHeightPx}px`,
                        width: '100%',
                        backgroundColor: '#202B33'
                    }}
                >
                    <NonIdealState
                        visual="timeline-line-chart"
                        title="No Measure Window Data Selected"
                    />
                </div>
            );
        }

        const stationConfig: StationConfig = {
            id: 'measure',
            name: 'measure',
            height: this.state.measureWindowHeightPx - TIME_AXIS_HEIGHT,
            defaultChannel: this.state.measureWindowSelection.channelRef.props.channelConfig,
            nonDefaultChannels: []
        };
        return (
            <div
                className="weavess-measure-window"
                ref={ref => this.containerRef = ref}
                style={{
                    flex: '0 0 auto',
                    borderBottom: '0.25rem solid',
                    height: `${this.state.measureWindowHeightPx}px`,
                    width: '100%',
                }}
            >
                <div
                    ref={ref => this.measureWindowRef = ref}
                    style={{
                        height: `${this.state.measureWindowHeightPx - TIME_AXIS_HEIGHT}px`,
                        width: '100%',
                        display: 'flex',
                    }}
                >
                    <StationLabel
                        ref={ref => this.stationLabelRef = ref}
                        selectedChannels={[]}
                        expanded={false}
                        stationConfig={stationConfig}
                        yAxisBounds={[{
                            channelId: this.state.measureWindowSelection.channelId ?
                                this.state.measureWindowSelection.channelId : '1',
                            bounds: this.state.yAxisBounds ?
                                    [this.state.yAxisBounds[0], this.state.yAxisBounds[1]] : [0, 1]
                        }]}
                        toggleExpansion={() => { /* no-op */ }}
                    />
                    <div
                        style={{
                            flex: '1 0 auto',
                            position: 'relative'
                        }}
                    >
                        <div
                            style={{
                                position: 'absolute',
                                top: '0px',
                                bottom: '0px',
                                right: '0px',
                                left: '0px',
                                overflowY: 'scroll'
                            }}
                        >
                            <Station
                                ref={ref => this.stationRef = ref}
                                stationConfig={stationConfig}
                                displayStartTimeSecs={this.state.measureWindowSelection.startTimeSecs}
                                displayEndTimeSecs={this.state.measureWindowSelection.endTimeSecs}
                                workerRpcs={this.props.workerRpcs}
                                selectedSignalDetections={this.props.selectedSignalDetections}
                                selectedChannels={[]}
                                expanded={false}
                                isMeasureWindow={true}
                                amplitudeScaleHotKey={this.props.amplitudeScaleHotKey}
                                amplitudeScaleSingleResetHotKey={this.props.amplitudeScaleSingleResetHotKey}
                                onSignalDetectionClick={this.props.onSignalDetectionClick}
                                onSignalDetectionDragEnd={this.props.onSignalDetectionDragEnd}
                                toggleExpansion={() => { /* no-op */ }}
                                setYAxisBounds={this.setYAxisBounds}
                                getViewRange={() => [0, 1]}
                                onContextMenu={this.props.onContextMenu}
                                onLoad={this.props.onLoad}
                                onKeyPress={this.props.onKeyPress}
                                onMouseMove={() => { /* no-op */ }}
                                onMouseDown={this.props.onMouseDown}
                                onMouseUp={this.props.onMouseUp}
                                updateMeasureWindow={() => { /* no-op */ }}
                                onMaskClick={this.props.onMaskClick}
                            />
                        </div>
                    </div>
                </div>
                <TimeAxis
                    startTimeSecs={this.state.measureWindowSelection.startTimeSecs}
                    endTimeSecs={this.state.measureWindowSelection.endTimeSecs}
                    getViewRange={() => [0, 1]}
                    borderTop={false}
                />
                <div
                    className="measure-window-divider"
                    onMouseDown={this.onMeasureWindowDividerDrag}
                    style={{
                        position: 'relative',
                        height: '0.5rem',
                        width: '100%',
                        cursor: 'row-resize',
                        flex: '0 0 auto'
                    }}
                >
                    <div
                        style={{
                            position: 'absolute',
                            top: '0.2rem',
                            bottom: '0.2rem',
                            left: '0px',
                            right: '0px',
                        }}
                    />
                </div>
            </div>
        );
    }

    /**
     * Get Channel Client bounds
     */
    public getChannelClientBounds = () => {
        if (!this.stationRef ||
            !this.stationRef.defaultChannelRef ||
            !this.stationRef.defaultChannelRef.containerRef) return;
        return this.stationRef.defaultChannelRef.containerRef.getBoundingClientRect();
    }

    /**
     * set the y-axis bounds for a particular channel
     */
    private readonly setYAxisBounds = (channelId: string, min: number, max: number) => {
        this.setState({
            yAxisBounds: [min, max]
        });
        this.props.onLoad();
    }

    /**
     * Update size of the measure window
     */
    private readonly onMeasureWindowDividerDrag = (e: React.MouseEvent<HTMLDivElement>) => {
        let prevPosition = e.clientY;
        let currentPos = e.clientY;
        let diff = 0;
        const minHeightPx = StationLabel.defaultProps.height;
        const maxHeightPx = 500;
        const onMouseMove = (e2: MouseEvent) => {
            if (this.containerRef && this.stationLabelRef) {
                currentPos = e2.clientY;
                diff = currentPos - prevPosition;
                prevPosition = currentPos;
                const heightPx = this.containerRef.clientHeight + diff;
                if (heightPx > minHeightPx && heightPx < maxHeightPx) {
                    this.setState({
                        measureWindowHeightPx: heightPx
                    });
                    this.props.onLoad();
                }
            }
        };

        const onMouseUp = (e2: MouseEvent) => {
            this.props.onLoad();
            document.body.removeEventListener('mousemove', onMouseMove);
            document.body.removeEventListener('mouseup', onMouseUp);
        };

        document.body.addEventListener('mousemove', onMouseMove);
        document.body.addEventListener('mouseup', onMouseUp);
    }
}
