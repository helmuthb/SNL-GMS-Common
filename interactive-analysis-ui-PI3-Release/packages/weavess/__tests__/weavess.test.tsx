import * as React from 'react';
import { WaveformDisplay, WaveformDisplayProps } from '../src/ts/components/waveform-display';
import { StationConfig, DisplayType,
    WeavessEvents, StationDefaultConfig, ChannelConfig, ChannelDefaultConfig, ChannelSegment,
    SignalDetectionConfig, TheoreticalPhaseWindow, Mask, WeavessSelections,
    HotKeyOverridesConfig, MeasureWindowSelection
} from '../src/ts/entities';
import {
    startTimeSeconds,
    endTimeSeconds
} from '../../../test/__test_runner_config__/analyst-ui-core/azimuth-slowness/test_util';

// function initialize(container: any, props: any): void {
// function initializeRecordSection(container: any, props: any): void {
// import { RecordSection } from '../src/ts/components/record-section-display';
// import { createDummyWaveform } from '../../weavess/src/ts/util/WaveformUtils';

const sampRate = 40;

const stationConfig: StationConfig[] = [
    // Beam
    {
        id: 'Beam',
        name: 'Beam',
        height: 75,
        defaultChannel: {
            id: 'Beam',
            name: 'Beam',
            color: 'dodgerblue',
            sampleRate: sampRate,
            dataSegments: [{
                startTimeSecs: startTimeSeconds,
                data: []
            }],
            displayType: [DisplayType.LINE],
            pointSize: 2,
        },
        nonDefaultChannels: []
    },
    // Fstat
    {
        id: 'Fstat',
        name: 'Fstat',
        height: 75,
        defaultChannel: {
            id: 'Fstat',
            name: 'Fstat',
            color: 'dodgerblue',
            sampleRate: sampRate,
            dataSegments: [{
                startTimeSecs: startTimeSeconds,
                data: []
            }],
            displayType: [DisplayType.LINE, DisplayType.SCATTER],
            pointSize: 2,
        },
        nonDefaultChannels: []
    },
    // Azimuth
    {
        id: 'Azimuth',
        name: 'Azimuth',
        height: 75,
        defaultChannel: {
            id: 'Azimuth',
            name: 'Azimuth',
            color: 'dodgerblue',
            sampleRate: sampRate,
            dataSegments: [{
                startTimeSecs: startTimeSeconds,
                data:[]
            }],
            displayType: [DisplayType.LINE, DisplayType.SCATTER],
            pointSize: 2,
        },
        nonDefaultChannels: []
    },
    // Slowness
    {
        id: 'Slowness',
        name: 'Slowness',
        height: 75,
        defaultChannel: {
            id: 'Slowness',
            name: 'Slowness',
            color: 'dodgerblue',
            sampleRate: sampRate,
            dataSegments: [{
                startTimeSecs: startTimeSeconds,
                data: []
            }],
            displayType: [DisplayType.LINE, DisplayType.SCATTER],
            pointSize: 2,
        },
        nonDefaultChannels: []
    }
];

const waveformDisplayProps: WaveformDisplayProps = {
    startTimeSecs: startTimeSeconds,
    endTimeSecs: endTimeSeconds,
    stations: stationConfig,
    defaultZoomWindow: {
        startTimeSecs: startTimeSeconds,
        endTimeSecs: endTimeSeconds
    },
    verticalMarkers: {
        color: 'dodgerblue',
        timeSecs: 10
    }[0],
    flex: false
};

// This is the actual test written for Jest
it('test to see if the waveform display renders correctly', () => {
  const wrapper = shallow(
    <div
        style={{
            height: '100%',
            width: '100%',
            display: 'flex',
        }}
    >
        <div
            style={{
                flex: '1 1 auto',
                position: 'relative'
            }}
        >
            <div
                style={{
                    position: 'absolute',
                    top: '0px', bottom: '0px', left: '0px', right: '0px',
                }}
            >
                < WaveformDisplay
                    {...waveformDisplayProps}
                />
            </div>
        </div>
    </div>
    );
  expect(wrapper).toMatchSnapshot();
});
