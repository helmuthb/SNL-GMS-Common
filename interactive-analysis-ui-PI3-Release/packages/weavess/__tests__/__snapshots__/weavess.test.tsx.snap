// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test to see if the waveform display renders correctly 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "height": "100%",
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "flex": "1 1 auto",
        "position": "relative",
      }
    }
  >
    <div
      style={
        Object {
          "bottom": "0px",
          "left": "0px",
          "position": "absolute",
          "right": "0px",
          "top": "0px",
        }
      }
    >
      <WaveformDisplay
        defaultZoomWindow={
          Object {
            "endTimeSecs": 1274320801,
            "startTimeSecs": 1274313599,
          }
        }
        endTimeSecs={1274320801}
        flex={false}
        startTimeSecs={1274313599}
        stations={
          Array [
            Object {
              "defaultChannel": Object {
                "color": "dodgerblue",
                "dataSegments": Array [
                  Object {
                    "data": Array [],
                    "startTimeSecs": 1274313599,
                  },
                ],
                "displayType": Array [
                  "LINE",
                ],
                "id": "Beam",
                "name": "Beam",
                "pointSize": 2,
                "sampleRate": 40,
              },
              "height": 75,
              "id": "Beam",
              "name": "Beam",
              "nonDefaultChannels": Array [],
            },
            Object {
              "defaultChannel": Object {
                "color": "dodgerblue",
                "dataSegments": Array [
                  Object {
                    "data": Array [],
                    "startTimeSecs": 1274313599,
                  },
                ],
                "displayType": Array [
                  "LINE",
                  "SCATTER",
                ],
                "id": "Fstat",
                "name": "Fstat",
                "pointSize": 2,
                "sampleRate": 40,
              },
              "height": 75,
              "id": "Fstat",
              "name": "Fstat",
              "nonDefaultChannels": Array [],
            },
            Object {
              "defaultChannel": Object {
                "color": "dodgerblue",
                "dataSegments": Array [
                  Object {
                    "data": Array [],
                    "startTimeSecs": 1274313599,
                  },
                ],
                "displayType": Array [
                  "LINE",
                  "SCATTER",
                ],
                "id": "Azimuth",
                "name": "Azimuth",
                "pointSize": 2,
                "sampleRate": 40,
              },
              "height": 75,
              "id": "Azimuth",
              "name": "Azimuth",
              "nonDefaultChannels": Array [],
            },
            Object {
              "defaultChannel": Object {
                "color": "dodgerblue",
                "dataSegments": Array [
                  Object {
                    "data": Array [],
                    "startTimeSecs": 1274313599,
                  },
                ],
                "displayType": Array [
                  "LINE",
                  "SCATTER",
                ],
                "id": "Slowness",
                "name": "Slowness",
                "pointSize": 2,
                "sampleRate": 40,
              },
              "height": 75,
              "id": "Slowness",
              "name": "Slowness",
              "nonDefaultChannels": Array [],
            },
          ]
        }
      />
    </div>
  </div>
</div>
`;
